doc/build.rst

```diff
@@ -71,15 +75,17 @@ Our goal is to build the shared library:
 - On Linux/OSX the target library is ``libxgboost.so``
 - On Windows the target library is ``xgboost.dll``
 
-The minimal building requirement is
+This shared library is used by different language bindings (with some additions depending
+on the binding you choose).  For building language specific package, see corresponding
+sections in this document.  The minimal building requirement is
 
 - A recent C++ compiler supporting C++11 (g++-5.0 or higher)
-- CMake 3.3 or higher (3.12 for building with CUDA)
+- CMake 3.12 or higher.
```

Comment:
Is cmake 3.12 strictly necessary?

Reply:
Since https://github.com/dmlc/xgboost/pull/5146, yes. It was done to support compiling XGBoost with OpenMP on Mac OSX.

Reply:
@hcho3 changed it to 3.12 for Apple Clang in `CMakeLists.txt`